apps/api/src/app/preferences/dtos/preferences.dto.ts

```diff
@@ -0,0 +1,42 @@
+import { Type } from 'class-transformer';
+import { IsBoolean, ValidateNested } from 'class-validator';
+
+class Preference {
+  @IsBoolean()
+  defaultValue: boolean;
+
+  @IsBoolean()
+  readOnly: boolean;
+}
+
+class Channels {
+  @ValidateNested({ each: true })
+  @Type(() => Preference)
+  in_app: Preference;
+
+  @ValidateNested({ each: true })
+  @Type(() => Preference)
+  email: Preference;
+
+  @ValidateNested({ each: true })
+  @Type(() => Preference)
+  sms: Preference;
+
+  @ValidateNested({ each: true })
+  @Type(() => Preference)
+  chat: Preference;
+
+  @ValidateNested({ each: true })
+  @Type(() => Preference)
+  push: Preference;
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
❓ **question:** Can these also be keyed by the enum?‏